File: apps/backend/user-service/src/main/java/com/aitest/user/entity/Menu.java
package com.aitest.user.entity;

import com.aitest.common.entity.BaseEntity;
import com.baomidou.mybatisplus.annotation.TableField;
import com.baomidou.mybatisplus.annotation.TableName;
import lombok.Data;
import lombok.EqualsAndHashCode;

/**
 * 菜单实体
 */
@Data
@EqualsAndHashCode(callSuper = true)
@TableName("menu")
public class Menu extends BaseEntity {

    /**
     * 父菜单ID
     */
    private Long parentId;

    /**
     * 菜单名称
     */
    private String name;

    /**
     * 路由路径
     */
    private String path;

    /**
     * 组件路径
     */
    private String component;

    /**
     * 图标
     */
    private String icon;

    /**
     * 类型: 0-菜单 1-按钮
     */
    private Integer type;

    /**
     * 权限标识
     */
    @TableField("permission_code")
    private String permission;

    /**
     * 排序
     */
    private Integer sortOrder;

    /**
     * 状态: 0-禁用 1-启用
     */
    private Integer status;
}

File: apps/frontend/src/api/menu.ts
import request from '@/utils/request'

export interface Menu {
  id: number
  name: string
  parentId: number
  path?: string
  component?: string
  icon?: string
  sortOrder?: number
  type: number
  permissionCode?: string
  status: number
  children?: Menu[]
  createTime?: string
  updateTime?: string
}

export interface MenuCreate {
  name: string
  parentId?: number
  path?: string
  component?: string
  icon?: string
  sortOrder?: number
  type?: number
  permissionCode?: string
}

export interface MenuUpdate {
  name?: string
  parentId?: number
  path?: string
  component?: string
  icon?: string
  sortOrder?: number
  type?: number
  permissionCode?: string
  status?: number
}

export const menuApi = {
  getTree: () => request.get<any>('/api/menus/tree'),

  list: () => request.get<any>('/api/menus'),

  get: (id: number) => request.get<any>(`/api/menus/${id}`),

  create: (data: MenuCreate) => request.post<any>('/api/menus', data),

  update: (id: number, data: MenuUpdate) => request.put<any>(`/api/menus/${id}`, data),

  delete: (id: number) => request.delete<any>(`/api/menus/${id}`),

  getPermissions: () => request.get<any>('/api/menus/permissions'),
}
